### DISCUSSION TITLE
Crossref Supplementary figures

### DISCUSSION THREAD

**Discussion by wolass** [quarto-dev/quarto-cli#8242] [Q&A] [answered]
### Description

As I am writing scientific papers in quarto, I need to have two figure counters and referencers: one for the figures that make it into the main paper and the second for secondary figures. It is a bit annoying to post process my manuscripts every time I re-render them to have accurate references to supplementary figures. These Supplementary figures usually start from 1 again, and I tend to have 6 Figs in the main body and additional 20 n the Supplement. 

I would really love to be able to reference my supplementary figures the same way the figs are now references with a custom label. 
like: @sfig-supplementaryfigure1. 

I thought this could be done using custom crossrefs but I understand this is yet to be implemented? 
[issue 8034](https://github.com/quarto-dev/quarto-cli/discussions/8034)

**Comment by mcanouil** [comment, marked as answer]
See <https://quarto.org/docs/authoring/cross-references-custom.html>.
Extensive documentation is on its way for the 1.4 release.

The discussion you linked is unrelated to your question.

**Reply by wolass** [reply]
Thank you for a rapid response!

**Comment by jamesprice202** [comment]
Probably a daft question but I can't get this to work...

In the YAML I've added:

crossref:
  custom:
    - kind: float
      key: suppfig
      latex-env: suppfig
      reference-prefix: Figure S
      space-before-numbering: false
      latex-list-of-description: Supplementary Figure

In the text I've written "see @suppfig-histogram for..."

And in the label of the source code I've written "@suppfig-histogram"

But the output just reads "see **?@suppfig-histogram**"

Any ideas? Thanks in advance

**Reply by mcanouil** [reply]
Did you look at the documentation and try to modify the example to suit your needs?
Which version of Quarto are you using?

Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.


Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by andjar** [comment]
Would it be possible to further expand the documentation on supplementary materials, e.g., how the cells should be set up with key and figure caption? It's an important feature and I'm really grateful to the team for having implemented it in Quarto!
